core/cas-server-core-events/src/test/java/org/apereo/cas/support/events/HttpServletRequestSimulation.java

```diff
@@ -0,0 +1,84 @@
+package org.apereo.cas.support.events;
+
+
+import org.apereo.cas.mock.MockTicketGrantingTicket;
+import org.apereo.cas.support.events.ticket.CasTicketGrantingTicketCreatedEvent;
+import org.apereo.cas.util.HttpRequestUtils;
+import lombok.val;
+import org.apereo.inspektr.common.web.ClientInfo;
+import org.apereo.inspektr.common.web.ClientInfoHolder;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.mock.web.MockHttpServletRequest;
+import java.util.concurrent.Callable;
+
+
+/**
+ *  This is {@link HttpServletRequestSimulation}.
+ *
+ *   This Callable class simulates a thread that an application server would create for
+ *   incoming HttpServletRequests, and will publish a CasTicketGrantingTicketCreatedEvent
+ *   to the application context.  It will populate the ClientInfo object for each 'thread' with data from the HttpServletRequest.
+ *   This is meant to be used for testing with the CasAuthenticationEventListenerTests class.
+ *   It will set the ips of the server/client to either IP1 or IP2 depending on the constructor value.
+ *
+ *  @author David Malia
+ *  @since 6.6
```

Comment:
Fix the version please.

Reply:
I set the version to 6.6.6.  I think that is the right version.  If not, what version should I use?